feat(Aside): standardized width size

laraharrow opened the pull request:
- [ ] ♿️ Accessibility addressed
- [ ] 🌏 Localization & Internationalization addressed
- [ ] 🖼 Image Snapshot coverage
- [ ] 📚 Documentation updated
- [ ] 🧳 Bundle size impact addressed
- [ ] 🏁 Performance impacts addressed
- [ ] 👾 Browsers tested
  - [ ] Chrome
  - [ ] Firefox
  - [ ] Safari
  - [ ] IE11

lukelooker reviewed (changes requested):
Need to add `AsideSizes` to the sizes supported by `Drawer` so that folks can move between the two seamlessly then this is totally good to go!

lukelooker commented on packages/components/src/Drawer/DrawerSurface.tsx:
```diff
@@ -75,6 +76,7 @@ export const DrawerSurface = styled(SurfaceBase).attrs<DrawerSurfaceProps>(
 
   ${placement}
   ${dialogWidth}
+  ${asideWidth}
```
This will output `width` twice in the CSS.

There should be a new `drawerWidth` function that combines the two Ramps and then allows a single use here.

lukelooker reviewed (changes requested):
Seeing some test failures and the `DrawerWidth` is going to need to be an extension of `DialogWidth` so there's only a single helper in the CSS block.

mdodgelooker commented on packages/components/src/Drawer/DrawerSurface.tsx:
```diff
@@ -43,7 +44,7 @@ export interface DrawerSurfaceProps {
    * @default 'medium'
    */
-  width?: ResponsiveValue<DialogSizeRamp | string>
+  width?: ResponsiveValue<DialogSizeRamp | AsideSizeRamp | string>
```
If you make this a new type...
```
export type DrawerWidth = ResponsiveValue<DialogSizeRamp | AsideSizeRamp | string>
...
width?: DrawerWidth
```
You can then add it to `UseDialogPropsInternal` in `useDialog.tsx`:
```
export interface UseDialogPropsInternal
  extends Omit<UseDialogProps, 'placement' | 'width'> {
  placement?: DialogPlacements | DrawerPlacements
  width?: DialogWidth | DrawerWidth
}
```

lukelooker commented on packages/components/src/Dialog/DialogSurface.tsx:
```diff
@@ -63,7 +64,7 @@ export interface DialogSurfaceProps {
    * the specified width or the viewport width.
    */
-  width?: DialogWidth
+  width?: DrawerWidth
```
?

```suggestion
  width?: DialogWidth
```

lukelooker commented on packages/components/src/Drawer/DrawerSurface.tsx:
```diff
@@ -43,7 +47,7 @@ export interface DrawerSurfaceProps {
    * @default 'medium'
    */
-  width?: ResponsiveValue<DialogSizeRamp | string>
+  width?: DrawerWidth
```
This needs to support `string` still as folks may want to use "off standard" sizes:

```suggestion
  width?: DrawerWidth | string
```

Also removing `ResponsiveValue` will act as a breaking change for anybody using that so we need to put it back so as not to break folks who might be using responsive drawer sizes (not sure if anyone is but it's hard to know for sure since extension developers use stuff in new ways).

lukelooker commented on packages/components/src/Dialog/DialogSurface.tsx:
```diff
@@ -28,7 +28,8 @@ import styled, { css } from 'styled-components'
 import { theme } from '@looker/design-tokens'
 import { SurfaceBase, surfaceTransition } from '../Dialog/SurfaceBase'
-import { dialogWidth, DialogWidth } from './dialogWidth'
+import { DrawerWidth } from '../Drawer/DrawerSurface'
```
I'm thinking the `DrawerWidth` shouldn't be pulled into `DialogSurface`?

lukelooker replied:
`DialogDrawerWidth`

lukelooker reviewed (changes requested):
Appears that sine `Drawer` stuff snuck in `Dialog` and I see one spot where a change to `width` on `Drawer` could/would be a breaking change for folks.

laraharrow commented:
Hey Luke, I'm not sure I follow your suggestions here. I was getting errors by using `DialogWidth` within `Drawer` and `Dialog ` so to fix it I updated that value to include the new `Aside` values. Nothings should have changed to `Dialog` as we still support all the value expected by it. I just added the values that `Aside` is adding. I was a bit skeptical of adding `DrawerWidth` inside `DialogSurface` but it seemed to be needed to avoid a bug since `Drawer` is actually using `Dialog` underneath.

lukelooker reviewed (approved):
Ship it! :)
